### DISCUSSION TITLE
Ordering of Transactions and date

### DISCUSSION THREAD

**Discussion by Retired99** [moneymanagerex/moneymanagerex#5160] [Q&A] [answered]
Previous to the .21 update, if there were more than on transaction on a date then the transactions would sort by ID number. Currently when I enter transaction #2 for the same date, #2 is listed before #1. This makes comparing the running balance on MMEX to the statement not possible. I have went back to the previous version (portable) and everything works fine. Any suggestions?

**Comment by whalley** [comment, marked as answer]
It's a known issue in the current release primarily due to the use of primary and secondary sort implementation. Will be corrected in the next release.

**Comment by ccwking** [comment]
I think my bug is the same as above, just needed to list it here to confirm.
If I have repeated transfers (DD etc) and they are added, the ID of the transaction and any manual entries added after this are not sorting and this can result on the view transactions out of sequence
So the top balance (on that date - i.e. 16th Oct 2022) would have them sorted incorrectly, this results in the top transaction on that date being the higher monetary value than the lowest entry which is the correct balance.

**Comment by whalley** [comment]
Feel free to revert to v1.5.20 for now or use the latest Beta - https://ci.appveyor.com/project/moneymanagerex/moneymanagerex/builds/45084700/job/ab413xk87fsi5906/artifacts

**Comment by ccwking** [comment]
Thanks, however I am running on an iMac, is there a beta for this?

**Comment by whalley** [comment]
Sure, try here - https://mmex.ipx.co.uk/

Betas for all platforms can be found here: https://moneymanagerex.org/docs/downloads/development/

**Reply by ccwking** [reply]
thanks, installed fine, the default sort is Date then ID - how do I get it to display in Date in the correct sequence is the IDs are different and dont make the transactions appear in the correct sequence?

**Reply by tactilis** [reply]
@ccwking 

Could you try clicking on another column header (e.g Payee) to sort by that, then clicking back on Date, which will now set the sort to Date (primary sort) plus ID (secondary sort).

Does that fix it please?

 ---
@whalley  I think we have seen a couple of other questions like this. I'm not sure whether there is an upgrade issue depending on the primary/secondary sort order the user had in 1.5.21 prior to updating. If, for example, they had Date|Payee, would this automatically change to Date|ID when running 1.6.0?

**Reply by ccwking** [reply]
Ill do some checking, there are a lot of transactions. I think when I first loaded it was Date (Down Arrow) ID - Up Arrow
It seems to be now they are both down arrows, but I need to check to see if the withdrawal amounts are now sorting correctly. 
Will update later, work gets in the way now.

**Reply by tactilis** [reply]
> I think when I first loaded it was Date (Down Arrow) ID - Up Arrow

This should never happen in v1.6.0. The ID column will always be sorted in the same order as Date.

If this was the case, then as I suggested above, we may have an upgrade issue.

> It seems to be now they are both down arrows

i.e. Date|ID descending. That is what is expected.
Note that you can click on the Date column header again to reverse to sort order to Date|ID ascending.

> but I need to check to see if the withdrawal amounts are now sorting correctly

It should be correct.
